[Aishwarya] iP

aishh12 opened the pull request:
## Blitz
> *"For every minute spent in organising, an hour is earned."*
> *-Benjamin Franklin* ([source](https://www.pinterest.com/pin/11962755246074205/))

Blitz is an **organization tool** that frees your mind of having to remember things you need to do. It is :

* **text-based**
* **easy** to learn
* **super fast** to use

Want to try it out? Just follow the steps below :

1. download it from [here](https://github.com/aishh12/ip/releases/download/A-Jar/ip.jar).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 🤗

And it's **FREE**!

Features:
- [x] Managing tasks ✔️
- [ ] Managing deadlines (coming soon) 👀
- [ ] Reminders (coming soon) 👀

The Blitz UI is loaded by invoking `run` in the `main` method :

```java
public static void main(String[] args) {
    new Blitz("data/blitz.txt").run();
}
```

Commit:
Add Gradle support

Commit:
greet,echo,exit

Commit:
Add,List

Commit:
mark task as done feature

Commit:
ToDos, Events, Deadlines

Commit:
Automated Text UI Testing

Commit:
handle errors

Commit:
add delete item feature

Commit:
update to conditional statements and coding style

Commit:
add save feature

Commit:
add dates and times feature

Commit:
add .gitignore

Commit:
Merge branch 'branch-Level-7'

Commit:
Merge branch 'branch-Level-8'

Commit:
add more OOP

Commit:
add organize into packages

Commit:
add JUnit Tests

Commit:
Add Jar File

Commit:
add JavaDoc

Commit:
follow the Coding Standard

Commit:
add Find feature

Commit:
Merge branch 'branch-A-JavaDoc'

Commit:
Merge branch 'branch-A-CodingStandard'

Commit:
Merge branch 'add-gradle-support' of https://github.com/aishh12/ip into branch-A-Gradle

Commit:
automate project builds using Gradle

Commit:
add gradle files

Commit:
Merge branch 'branch-A-Gradle' into branch-A-CheckStyle

qreoct commented on src/main/java/TaskList.java:
```diff
@@ -0,0 +1,55 @@
+        }
+    }
+
+    public TaskList findMatchingTasks(String keyword) {
```
Minor point here, but I felt that the name `findMatchingTasks` might be better written as `findTasksThatMatchKeyword`. This is because `findMatchingTasks` makes me think that the code will find and group `Task`s based on their similarities, instead of a method that queries for `Task`s based on a given String.

qreoct commented on src/main/java/Ui.java:
```diff
@@ -0,0 +1,45 @@
+    public void printList(TaskList tasks, String str) {
+        int ctr = 1;
+        System.out.println("\t" + str);
+        //System.out.println("\tHere are the tasks in your list:");
```
Commented code can be removed since it is no longer needed.

qreoct commented on src/main/java/Parser.java:
```diff
@@ -0,0 +1,170 @@
+                        throw new BlitzException("No matches found!");
+                    }
+
+                    ui.printLine();
+                    ui.printList(matchingList, "Here are the matching tasks in your list:");
+                    ui.printLine();
```
Perhaps you could rewrite the `Ui::printFormatted` method to handle `TaskList`s. It would make the code cleaner.

qreoct reviewed (commented):
I enjoyed reading your code! I've only found very minor stylistic tweaks, so keep up the good work! All the best~

Commit:
add CheckStyle

Commit:
add gradle files

Commit:
Merge branch 'branch-A-Gradle' into branch-Level-10

lowkangn commented on src/main/java/Blitz.java:
```diff
@@ -0,0 +1,55 @@
+
+import java.io.IOException;
+
+public class Blitz {
```
Missing JavaDoc header comments for many of the public classes and constructors, hopefully you're in the process of writing them up! :blush:

lowkangn commented on src/main/java/Parser.java:
```diff
@@ -0,0 +1,170 @@
+     * @param task task to be deleted or marked as done (required).
+     * @return index of the required task in the task list.
+     */
+    private static int getTaskNumber(String keyword, String task) {
```
JavaDoc header comments are actually not needed for private methods, just thought you might want to know

lowkangn commented on src/main/java/Parser.java:
```diff
@@ -0,0 +1,170 @@
+     * @return index of the required task in the task list if valid.
+     * @throws BlitzException if the computed index is invalid.
+     */
+    public static int getIndex(String command, int listSize) throws BlitzException {
```
Same for getters/setters, you actually don't need JavaDoc header comments

lowkangn commented on src/main/java/Parser.java:
```diff
@@ -0,0 +1,170 @@
+     *     if a request to print an empty list is made or if the user enters
+     *     an invalid command.
+     */
+    public static void parse(String command, TaskList tasks, Ui ui) throws BlitzException{
```
Your parse method is really long. Maybe you could try to abstract some of the code out into helper methods?

lowkangn commented on src/main/java/Parser.java:
```diff
@@ -0,0 +1,170 @@
+                        throw new BlitzException("Incorrect date/time format! Please enter "
+                                + "event date/time in \"d M yyyy HHmm\" format");
+                    }
+                    break;
```
Lack of a default statement

lowkangn commented on src/main/java/Storage.java:
```diff
@@ -0,0 +1,96 @@
+                    current = new Deadline(keywords[2], LocalDateTime.parse(keywords[3],
+                            DateTimeFormatter.ofPattern("MMM d yyyy hh:mm a")));
+                    break;
+                default :
```
Try to have the default statement be used for the intended default option, and not just to execute the last option.
> If there is no default action, you can use the default branch to detect errors.

lowkangn commented on src/main/java/Storage.java:
```diff
@@ -0,0 +1,96 @@
+        FileWriter fw = new FileWriter("data/blitz.txt");
+        for(Task t : tasks.getTaskList()) {
+            if(t instanceof Todo) {
+                fw.write("T | " + t.getStatusIcon() + " | " + t.getDescription()
```
I think it would be cleaner and more readable if this processing was abstracted out

lowkangn reviewed (commented):
A good job overall! :smiley: There wasn't any major violations of code quality or the Java coding standard. There were some areas that could be improved but they can be resolved pretty easily.

kaldius commented on src/main/java/Parser.java:
```diff
@@ -0,0 +1,170 @@
+                default:
+                    String findKeyword = keywords[1];
+
+                    if(tasks.size() == 0) {
```
Just a small coding standard violation: `if` statements should be followed with a whitespace.

kaldius commented on src/main/java/Ui.java:
```diff
@@ -0,0 +1,45 @@
+    }
+
+    public void printList(TaskList tasks, String str) {
+        int ctr = 1;
```
It seems the variable `ctr` is not used in this method, it can be removed.

kaldius commented on src/main/java/tests/TaskListTest.java:
```diff
@@ -0,0 +1,90 @@
+import tasks.Task;
+import tasks.Todo;
+
+import java.time.LocalDateTime;
```
Imports from the java standard package should be placed above custom package imports.

kaldius commented on src/main/java/Parser.java:
```diff
@@ -0,0 +1,170 @@
+                }
+            } else {
+                Task current = new Task("");
+                switch (firstKeyword) {
```
Just a suggestion: this part of your control flow logic could be placed in an abstract `Task` class. `Task` could have a static factory method that checks the type of task to create then creates the necessary type. That should help in making this `parse` method shorter!

kaldius reviewed (commented):
Hey there! I had a good time reading your code. I found a few instances where `if` or `for` statements weren't followed by a whitespace but otherwise I think your code follows the coding standard pretty well! I also added a few small suggestions that might make your code a little neater/easier to read. Overall I think this is great work! All the best!

Commit:
Merge branch 'branch-A-CheckStyle' into branch-Level-10

Commit:
add GUI

Commit:
add Jar file using gradle

Commit:
Add java assertions

There are no checks for important assumptions that should hold at various points in the code.

Lack of check for these key assumptions could lead to possible bugs.

Adding assertions in places where the rest of the program depends on a user defined function
allows us to verify that we have desirable program states throughout.

Commit:
Add improvements to code quality

Method to parse the inputted user command was rather lengthy and deeply nested.

Very long and deeply methods reduce readability and code quality and indicate the
need for better abstraction.

TaskList related methods have been moved to the TaskList class while common parts
of the code in switch case branches and exception handling have been abstracted out.
Default cases have been included as well.

aishh12 committed:
Merge pull request #2 from aishh12/branch-A-CodeQuality

Add improvements to code quality

Commit:
Merge branch 'master' of https://github.com/aishh12/ip

Commit:
Merge branch 'master' into branch-A-Assertions

aishh12 committed:
Merge pull request #1 from aishh12/branch-A-Assertions

Add java assertions

Commit:
Merge branch 'master' of https://github.com/aishh12/ip

Commit:
Add DetectDuplicates extension

Commit:
add improvements to GUI and help feature

Commit:
add edits to Ui.java

Commit:
add edit to Ui.java

aishh12 committed:
Set theme jekyll-theme-midnight

Commit:
add Ui.png

Commit:
add pictures for UG

aishh12 committed:
Update README.md

aishh12 committed:
Set theme jekyll-theme-hacker

aishh12 committed:
Update README.md

aishh12 committed:
Update README.md

aishh12 committed:
Update README.md

aishh12 committed:
Update README.md

aishh12 committed:
Update README.md

aishh12 committed:
Update README.md

Commit:
add edit to build.gradle

Commit:
Merge branch 'master' of https://github.com/aishh12/ip

Commit:
add edit to build.gradle

Commit:
edits to build.gradle
